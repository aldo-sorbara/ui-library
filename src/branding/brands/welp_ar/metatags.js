export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
