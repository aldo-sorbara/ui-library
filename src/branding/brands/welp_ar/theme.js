import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
