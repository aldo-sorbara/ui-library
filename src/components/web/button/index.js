export { Button, ButtonContainer } from './Button';
