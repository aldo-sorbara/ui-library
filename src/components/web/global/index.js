export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
