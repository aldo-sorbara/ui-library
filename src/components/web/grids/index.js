export { Grid, Container as GridContainer } from './Grids';
