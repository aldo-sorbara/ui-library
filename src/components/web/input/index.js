export { default, InputLabel, InputDescription } from './Input';
