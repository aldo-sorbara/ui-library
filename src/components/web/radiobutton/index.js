export { RadioButton, RadioGroup } from './Radio';
