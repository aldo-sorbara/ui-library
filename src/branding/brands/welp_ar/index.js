import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
