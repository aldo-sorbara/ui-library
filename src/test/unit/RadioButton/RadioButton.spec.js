import React from 'react';
import { RadioButton, RadioGroup } from '../../../components/web/radiobutton';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import 'jest-styled-components';

describe('RadioButton component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <RadioGroup
          className='test-radio'
          name='test'
          onChange={value => {
            console.log(value);
          }}
        >
          <RadioButton data='test1' value='test1' label='test1' />
          <RadioButton data='test2' value='test2' label='test2' />
        </RadioGroup>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render with default props', () => {
    const wrapper = mountWithTheme(
      <RadioGroup
        className='test-radio'
        name='test'
        onChange={value => {
          console.log(value);
        }}
      >
        <RadioButton data='test1' value='test1' label='test1' />
        <RadioButton data='test2' value='test2' label='test2' />
      </RadioGroup>,
      welp_ar.theme
    );
    expect(wrapper).toHaveStyleRule('display', 'block');
  });
  it('should render with selected prop by default', () => {
    const wrapper = mountWithTheme(
      <RadioButton selected data='test1' value='test1' label='test1' />,
      welp_ar.theme
    );
    expect(wrapper.props().selected).toBe(true);
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '0 3px 14px 0 rgba(21,67,118,0.12)'
    );
    expect(wrapper).toHaveStyleRule('border', '1px solid #c346f9');
  });
  it('should render null - not valid children element', () => {
    const wrapper = mountWithTheme(
      <RadioGroup
        className='test-radio'
        name='test'
        onChange={value => {
          console.log(value);
        }}
      >
        test
      </RadioGroup>,
      welp_ar.theme
    );
    expect(wrapper.props().children).toBe('test');
  });
  it('should handle onChange func', () => {
    const onChange = jest.fn();
    const wrapper = mountWithTheme(
      <RadioGroup className='test-radio' name='test' onChange={onChange}>
        <RadioButton data='test1' value='test1' label='test1' />
        <RadioButton data='test2' value='test2' label='test2' />
      </RadioGroup>,
      welp_ar.theme
    );
    expect(
      wrapper
        .find('input')
        .at(0)
        .simulate('change')
    );
    expect(onChange).toHaveBeenCalledTimes(1);
  });
  it('should render inline radioButtons', () => {
    const wrapper = mountWithTheme(
      <RadioButton inline data='test2' value='test2' label='test2' />,
      welp_ar.theme
    );
    expect(wrapper.props().inline).toBe(true);
    expect(wrapper).toHaveStyleRule('display', 'inline-block');
    expect(wrapper).toHaveStyleRule('margin-right', '0.5rem');
  });
});
