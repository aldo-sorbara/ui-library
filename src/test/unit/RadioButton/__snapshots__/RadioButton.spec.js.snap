// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should render snapshot 1`] = `
.c0 {
  display: block;
}

.c0 input {
  left: 0;
  opacity: 0;
  position: absolute;
  top: 0;
}

.c0 span {
  background-color: #ffffff;
  border: 1px solid #0082ff;
  border-radius: 50%;
  height: 1rem;
  position: absolute;
  top: 1.125rem;
  -webkit-transition: border-color,background-color 0.2s ease-in-out;
  transition: border-color,background-color 0.2s ease-in-out;
  width: 1rem;
}

.c0 input:checked ~ span {
  background-color: #0082ff;
  border-color: #0082ff;
}

.c0 input:checked ~ span:after {
  display: block;
}

.c2 {
  color: inherit;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.25;
  padding: 1rem 1rem 1rem 2.5rem;
}

.c1 {
  background-color: #ffffff;
  border: 1px solid rgba(29,44,81,0.15);
  display: block;
  margin-bottom: .5rem;
  margin-right: 0;
  position: relative;
  -webkit-transition: border-color,background-color 0.2s ease-in-out;
  transition: border-color,background-color 0.2s ease-in-out;
}

.c1 label {
  padding: 1rem 1rem 1rem 2.5rem;
}

.c1 span {
  left: 1rem;
}

<div
  className="c0"
>
  <div
    className="c1"
    selected={false}
  >
    <label
      className="c2"
    >
      <input
        checked={false}
        data-w="test1"
        name="test"
        onChange={[Function]}
        type="radio"
        value="test1"
      />
      <span />
      test1
    </label>
  </div>
  <div
    className="c1"
    selected={false}
  >
    <label
      className="c2"
    >
      <input
        checked={false}
        data-w="test2"
        name="test"
        onChange={[Function]}
        type="radio"
        value="test2"
      />
      <span />
      test2
    </label>
  </div>
</div>
`;
